src/index.ts

```diff
@@ -180,411 +77,142 @@ export function resolve(
     }
   }
 
-  initMappers(cachedOptions)
+  if (!resolver) {
+    const optionsHash = stableHash(options)
+    options = normalizeOptions(options)
```

Comment:
Should we hash options after normalization?

Reply:
There's already cache check, and there is no heavy task inside, so I think hash check could be redundant and unnecessary maybe.

Reply:
Will, both hash and normalize can't be skipped. So it won't make any performance differences.

Reply:
`normalize` is needed to determine whether only one tsconfig is used or there are 2+ projects, if projects > 1, we'll need to use `get-tsconfig` to check which project should be applied for input file. The performance would be different with 1 project vs 2+ projects.

That's why I added a warning about more than one project used, a union project entry with `references` would be better choice so that we can leverage `oxc-resolver`'s power instead of resolving by ourselves.

Reply:
Make sense then.